pkg/resources/management.cattle.io/v3/clusterproxyconfig/validator_test.go

```diff
@@ -0,0 +1,96 @@
+package clusterproxyconfig
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	v3api "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/webhook/pkg/admission"
+	wranglerfake "github.com/rancher/wrangler/pkg/generic/fake"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	admissionv1 "k8s.io/api/admission/v1"
+	authenicationv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	cpcGVR = metav1.GroupVersionResource{Group: "management.cattle.io", Version: "v3", Resource: "clusterproxyconfigs"}
+	cpcGVK = metav1.GroupVersionKind{Group: "management.cattle.io", Version: "v3", Kind: "ClusterProxyConfig"}
+)
+
+func Test_admitter_Admit(t *testing.T) {
+	tests := []struct {
+		name          string
+		alreadyExists bool
+		allowed       bool
+		wantErr       bool
+	}{
+		{
+			name:          "create clusterproxyconfig when none exists",
+			allowed:       true,
+			alreadyExists: false,
+		},
+		{
+			name:          "attempt to make more than one clusterproxyconfig",
+			allowed:       false,
+			alreadyExists: true,
+		},
+		{
+			name:    "failed to list clusterproxyconfigs",
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			ctrl := gomock.NewController(t)
+			cpsCache := wranglerfake.NewMockCacheInterface[*v3api.ClusterProxyConfig](ctrl)
+			cpsCache.EXPECT().List(gomock.Any(), gomock.Any()).DoAndReturn(func(namespace string, _ labels.Selector) ([]*v3api.ClusterProxyConfig, error) {
```

Comment:
nit: we could be more specific here instead of using `gomock.Any()`. We could verify the `namespace` and `labels.Everything()` are passed as arguments

Reply:
Ok, that's reasonable.